### ISSUE TITLE
Fatal error on the Products report page after deleting a variation

### ISSUE THREAD

**Issue by madeincosmos** [woocommerce/woocommerce-admin#5256]
Looks related to #4421 but the root cause here is different.

**Describe the bug**
**Analytics > Products** report for variable product gets broken when:

- there's more than 1 page of results in the report, and
- one of the variations gets deleted after someone places an order for it. 

The fatal error I'm seeing in this case:

```
2020-10-02T07:45:15+00:00 CRITICAL Uncaught Error: Call to a member function get_variation_attributes() on bool in /home/dellapia/public_html/wp-content/plugins/woocommerce/packages/woocommerce-admin/src/API/Reports/Variations/DataStore.php:207
Stack trace:
#0 /home/dellapia/public_html/wp-content/plugins/woocommerce/packages/woocommerce-admin/src/API/Reports/Variations/DataStore.php(334): Automattic\WooCommerce\Admin\API\Reports\Variations\DataStore->include_extended_info(Array, Array)
#1 /home/dellapia/public_html/wp-content/plugins/woocommerce/includes/class-wc-data-store.php(207): Automattic\WooCommerce\Admin\API\Reports\Variations\DataStore->get_data(Array)
#2 /home/dellapia/public_html/wp-content/plugins/woocommerce/packages/woocommerce-admin/src/API/Reports/Variations/Query.php(46): WC_Data_Store->__call('get_data', Array)
#3 /home/dellapia/public_html/wp-content/plugins/woocommerce/packages/woocommerce-admin/src/API/Reports/Variations/Controller.php(70): Automattic\WooCommerce\Admin\API\Reports\Variations\Query->get_data()
#4 /home/del in /home/dellapia/public_html/wp-content/plugins/woocommerce/packages/woocommerce-admin/src/API/Reports/Variations/DataStore.php on line 207
```

**To Reproduce**

1. Create a variable product with several variations.
2. Place several orders for it until there's more than 25 sales of this product total (these can be 1 order for many variations, or many orders of the same variation).
3. Delete one of the variations that were previously purchased.
4. Go to **Analytics > Products**, choose that variable product in the report and observe the Variations table isn't loading at all.

**Expected behavior**
Reports work even after I delete some variation. If it's not possible to obtain variation data, it could say 'unknown variation' or put a similar placeholder.

**Screenshots**
Less than 25 sales - the report is loading fine even after I delete some variation:

![Screenshot](https://d.pr/i/sTO59g+)
Screenshot: https://d.pr/i/sTO59g

More than 25 sales - the bottom part is stuck on endless loading:

![Screenshot](https://d.pr/i/xDdQ9A+)
Screenshot: https://d.pr/i/xDdQ9A

**Desktop (please complete the following information):**

* OS: MacOS 10.15.6
* Browser Chrome 83

**Additional context**
This error only happens when there's more than 1 page of results - otherwise we're exiting the `get_data` function here: 

https://github.com/woocommerce/woocommerce-admin/blob/master/src/API/Reports/Variations/DataStore.php#L309:L311

I think this might be fixed by changes proposed in this commit, but I'm not quite sure how I can build this version to test it:

https://github.com/woocommerce/woocommerce-admin/commit/372bd07417c0f094fb15c49ba27d2ee24693224a

**Comment by pmcpinto** [comment]
cc @jeffstieler

**Comment by timmyc** [comment]
Fixed in #5302 - thanks again for porting the original fix to the release branch @jeffstieler  🙇
